### ISSUE TITLE
Output and trajectory files don't match

### ISSUE THREAD

**Issue by KEggensperger** [automl/SMAC3#39]
When running rosenbrock.py I get the output shown below. The output suggests that there are at least 4 incumbents (not even listing the initial incumbent), but the trajectory file only lists three (including the incumbent). Which one is correct?

```
Default Value: 120048.00
/home/eggenspk/virtualEnvs/virtual3SMAC/lib/python3.4/site-packages/smac-0.1.0-py3.4.egg/smac/smbo/acquisition.py:173: RuntimeWarning: divide by zero encountered in true_divide
INFO:intensifier:Challenger (69492.7391) is better than incumbent (120048.0000) on 1 runs.
INFO:intensifier:Changing incumbent to challenger: Configuration:
  x1, Value: -4.656907303634218
  x2, Value: -3
  x3, Value: 2
  x4, Value: -2

INFO:intensifier:Challenger (677.3422) is better than incumbent (69492.7391) on 1 runs.
INFO:intensifier:Changing incumbent to challenger: Configuration:
  x1, Value: -0.8297799529742598
  x2, Value: 2
  x3, Value: 2
  x4, Value: 3

INFO:intensifier:Updated estimated performance of incumbent on 1 runs: 677.3422
INFO:intensifier:Updated estimated performance of incumbent on 1 runs: 677.3422
INFO:intensifier:Updated estimated performance of incumbent on 1 runs: 677.3422
INFO:intensifier:Challenger (511.8286) is better than incumbent (677.3422) on 1 runs.
INFO:intensifier:Changing incumbent to challenger: Configuration:
  x1, Value: -0.23478815616176085
  x2, Value: 0
  x3, Value: -2
  x4, Value: 5

INFO:intensifier:Challenger (410.7729) is better than incumbent (511.8286) on 1 runs.
INFO:intensifier:Changing incumbent to challenger: Configuration:
  x1, Value: 0.17154408126871612
  x2, Value: 0
  x3, Value: -2
  x4, Value: 4

INFO:intensifier:Updated estimated performance of incumbent on 1 runs: 410.7729
INFO:intensifier:Updated estimated performance of incumbent on 1 runs: 410.7729
INFO:intensifier:Updated estimated performance of incumbent on 1 runs: 410.7729
INFO:intensifier:Updated estimated performance of incumbent on 1 runs: 410.7729
INFO:intensifier:Updated estimated performance of incumbent on 1 runs: 410.7729
INFO:Stats:##########################################################
INFO:Stats:Statistics:
INFO:Stats:#Target algorithm runs: 204
INFO:Stats:Used wallclock time: 7.05 sec
INFO:Stats:Used target algorithm runtime: 0.55 sec
INFO:Stats:##########################################################
Final Incumbent: Configuration:
  x1, Value: 0.17154408126871612
  x2, Value: 0
  x3, Value: -2
  x4, Value: 4

Optimized Value: 410.77
```

The trajectory file looks like that:

```
"CPU Time Used","Estimated Training Performance","Wallclock Time","Incumbent ID","Automatic Configurator (CPU) Time","Configuration..."
0.004036, 999999999.000000, 0.012154, 1, 0.008118, x1='5.0', x2='5', x3='5', x4='5'
0.034413, 677.342200, 0.387002, 3, 0.352589, x1='-0.8297799529742598', x2='2', x3='2', x4='3'
0.210829, 410.772936, 2.789726, 5, 2.578898, x1='0.17154408126871612', x2='0', x3='-2', x4='4'
```

**Comment by mlindauer** [comment]
(similar to checking the configuration budgets)
the trajectory file is only updated after each iteration. 
Since we run at least two (in this example probably even more) configurations in each iterations,
it can happen that we change the incumbent more than once per iteration and the trajectory is updated only once.

We could pass the `TrajLogger` object also to `Intensifier`, but I would like to avoid this because the class has already quite a lot to handle.

However, I found another bug related this issue.
We first checked whether the configuration budget was empty and updated the trajectory file afterwards if the configuration was not empty. Therefore, we would not save the last incumbent in the trajectory file.
